cwltool/load_tool.py

```diff
@@ -195,21 +195,24 @@ def _add_blank_ids(workflowobj):
         for entry in workflowobj:
             _add_blank_ids(entry)
 
-def validate_document(document_loader,  # type: Loader
-                      workflowobj,  # type: CommentedMap
-                      uri,  # type: Text
-                      enable_dev=False,  # type: bool
-                      strict=True,  # type: bool
-                      preprocess_only=False,  # type: bool
+def validate_document(document_loader,           # type: Loader
+                      workflowobj,               # type: CommentedMap
+                      uri,                       # type: Text
+                      enable_dev=False,          # type: bool
+                      strict=True,               # type: bool
+                      preprocess_only=False,     # type: bool
                       fetcher_constructor=None,  # type: FetcherConstructorType
-                      skip_schemas=None,  # type: bool
-                      overrides=None,  # type: List[Dict]
-                      metadata=None,  # type: Optional[Dict]
-                      do_validate=True
+                      skip_schemas=None,         # type: bool
+                      overrides=None,            # type: List[Dict]
+                      metadata=None,             # type: Dict[Text, Any]
```

Comment:
This type should still be marked `optional` if the default value is `None`. This is enforced by newer versions of `mypy`